Automattic/newspack-newsletters
# Fix/non public newsletters in search results new

### All Submissions:






### Changes proposed in this Pull Request:

Newspack Newsletters were showing up in search results regardless of whether they were marked as Public or non-Public. This fix attempts to correct that behavior by ensuring that non-public posts do not show up in search results.



Closes #313 .

### How to test the changes in this Pull Request:

1. Create new newsletter.
2. Mark as sent. Do not make public.
3. Go to site search, and search for some content which is in the newsletter you created.
4. You should not see the newsletter in the search results.
5. Go back and make the newsletter public.
6. Go to site search again and search for some content which is in the newsletter.
7. You should see the newsletter in the search results.
8. Create a regular post and publish.
9. Search for content within that post.
10. You should not see that post in the search results.

*Negative testing*

1. create a post and publish.
2. For that post ID, add a new record to postmeta table with key = is_public and value = 1
3. Search for content within the post you created
4. The post should not be visible despite having the target content.
5. Create a post and publish
6. For that post ID, add a new record to postmeta table with key = is_public
7. Search for content within the post you created
8. The post should not be visible.

### Other information:
